### PR TITLE
update TORCH_LOGS_OUT to overwrite existing file instead of appending

### PR DESCRIPTION
**Pull request by bdhirsh**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118647
* #118646
* #118645
* #118644
* #118643
* __->__ #118937

### CODE REVIEW

**Review by oulgen — APPROVED** [review]
When I added this, I was actually taking advantage of the append behavior but overwrite is probably better for this feature overall.

File: torch/_logging/_internal.py [hunk 1]
```diff
@@ -831,7 +831,7 @@ def _init_logs(log_file_name=None):
 
         if log_file_name is not None:
             _setup_handlers(
-                lambda: logging.FileHandler(log_file_name),
+                lambda: logging.FileHandler(log_file_name, 'w+'),
```
Comment (albanD) [inline torch/_logging/_internal.py:834]: I would agree with Geeta that you most likely want to have both options.
I don't know what are the precedents for this but I could see `TORCH_LOGS_OUT=foo.txt` to override and `TORCH_LOGS_OUT=+foo.txt` to appends. Using the fact that a path starting with + is not valid.

Comment (bdhirsh) [reply torch/_logging/_internal.py:834]: Oh I like this idea - happy to switch to that (@oulgen does that sound good to you?)

Comment (oulgen) [reply torch/_logging/_internal.py:834]: Sounds good. Feel free to implement it, otherwise I can do it when I’m back from pto

Comment (bdhirsh) [reply torch/_logging/_internal.py:834]: will do in this PR!

**Pull request closed without merging**